src/sagemaker_tensorflow_container/training.py

```diff
@@ -90,27 +90,27 @@ def host_addresses(hosts, port=2222):
 
 def _env_vars_with_tf_config(env, ps_task):
     env_vars = env.to_env_vars()
-    env_vars['TF_CONFIG'] = json.dumps(_build_tf_config(
-        hosts=env.hosts,
-        current_host=env.current_host,
-        ps_task=ps_task))
+    tf_config = _build_tf_config(hosts=env.hosts,
+                                 current_host=env.current_host,
+                                 ps_task=ps_task)
+
+    env_vars['TF_CONFIG'] = json.dumps(tf_config, sort_keys=True)
     return env_vars
 
 
 def _run_ps(env):
-    env_vars = _env_vars_with_tf_config(env, ps_task=True)
-    # Parameter server processes should always run on CPU. Sets CUDA_VISIBLE_DEVICES to '-1' forces
-    # TensorFlow to use CPU.
-    env_vars['CUDA_VISIBLE_DEVICES'] = json.dumps(-1)
-    framework.entry_point.run(env.module_dir, env.user_entry_point,
-                              env.to_cmd_args(), env_vars, wait=False)
+    logger.info('Running distributed training job with parameter servers')
+
+    tf_config = _build_tf_config(hosts=env.hosts, current_host=env.current_host)
```

Comment:
should we just remove the ps_task parameter in  _build_tf_config? It's not used anymore. Not critical. if you don't want to fiddle with the unit tests we can do it later.